src/app/child-dev-project/attendance/dashboard-widgets/attendance-week-dashboard/attendance-week-dashboard.component.ts

```diff
@@ -68,92 +62,72 @@ export class AttendanceWeekDashboardComponent
-      this.attendanceRecordsMap.set(attendanceMonth.student, record);
+      rows
+        .filter((r) => this.filterLowAttendance(r))
+        .forEach((r) => lowAttendanceCases.add(r.childId));
     }
 
-    const relevantDays = attendanceMonth.dailyRegister.filter(
-      (a) =>
-        a.date.getTime() >= startDate.getTime() &&
-        a.date.getTime() <= endDate.getTime()
-    );
+    const groupedRecords = groupBy(records, "childId");
+    this.dashboardRowGroups = Array.from(
+      lowAttendanceCases.values()
+    ).map((childId) => groupedRecords.get(childId));
+  }
 
-    if (attendanceMonth.institution === "coaching") {
-      record.attendanceCoaching = record.attendanceCoaching
-        .concat(relevantDays)
-        .sort(
-          (a, b) =>
-            (a.date ? a.date.getTime() : 0) - (b.date ? b.date.getTime() : 0)
-        );
-    } else if (attendanceMonth.institution === "school") {
-      record.attendanceSchool = record.attendanceSchool
-        .concat(relevantDays)
-        .sort(
-          (a, b) =>
-            (a.date ? a.date.getTime() : 0) - (b.date ? b.date.getTime() : 0)
-        );
+  private generateRowsFromActivityAttendance(
+    att: ActivityAttendance,
+    from: Moment,
+    to: Moment
+  ): AttendanceWeekRow[] {
+    if (!att.activity) {
+      return [];
     }
-  }
 
-  private filterAbsentees() {
-    this.attendanceRecords = [];
+    const results: AttendanceWeekRow[] = [];
+    for (const participant of att.activity.participants) {
+      const eventAttendances = [];
 
-    this.attendanceRecordsMap.forEach((record) => {
-      const countAbsencesFun = (acc, v) =>
-        v.status === AttendanceStatus.ABSENT ? acc + 1 : acc;
-      record.schoolAbsences = record.attendanceSchool.reduce(
-        countAbsencesFun,
-        0
-      );
-      record.coachingAbsences = record.attendanceCoaching.reduce(
-        countAbsencesFun,
-        0
-      );
-      if (record.schoolAbsences > 1 || record.coachingAbsences > 1) {
-        this.attendanceRecords.push(record);
+      let day = moment(from);
+      while (day.isSameOrBefore(to, "day")) {
+        const event = att.events.find((e) => day.isSame(e.date, "day"));
+        if (event) {
+          eventAttendances.push(event.getAttendance(participant));
+        } else {
+          // put a "placeholder" into the array for the current day
+          eventAttendances.push(undefined);
+        }
+        day = day.add(1, "day");
       }
-    });
+
+      results.push({
+        childId: participant,
+        activity: att.activity,
+        attendanceDays: eventAttendances,
+      });
+    }
+
+    return results;
   }
 
-  goToChild(childId: string) {
-    this.router.navigate(["/child", childId]);
+  private filterLowAttendance(row: AttendanceWeekRow): boolean {
+    const countAbsences = row.attendanceDays.filter(
+      (e) => e?.status?.countAs === AttendanceLogicalStatus.ABSENT
+    ).length;
+
+    return countAbsences > 1;
```

Comment:
I think what is considered a _low attendance_ should be saved in a static variable and potentially be configurable. This `1` is a little bit magical for me.

Reply:
good point :blush: 

I could argue that "this is how the previous implementation was doing it already" but then there is the rule to always leave a place a bit cleaner than you found it. Will make this configurable.